src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -339,14 +340,21 @@ private void doRetrieve(SCMHeadObserver observer, TaskListener listener, GHRepos
                 return 0; // TODO
             }
             @Override public boolean exists(@Nonnull String path) throws IOException {
-                try {
-                    repo.getFileContent(path, ref);
-                    listener.getLogger().format("      %s exists in this %s%n", path, thing);
-                    return true;
-                } catch (FileNotFoundException x) {
-                    listener.getLogger().format("      %s does not exist in this %s%n", path, thing);
-                    return false;
+                List<GHContent> directoryContent = repo.getDirectoryContent("/", ref);
```

Comment:
:bug: this will work for a `path` like `Jenkinsfile`, but not like (say) `.jenkins/script.groovy`.

Assuming we only care about reporting on case mismatches for the final path component (so `Script.groovy` in the above example rather than `.Jenkins`), it would probably suffice to write something along the lines of

```java
int index = path.lastIndexOf('/') + 1;
List<GHContent> directoryContent = repo.getDirectoryContent(path.substring(0, index), ref); // impl removes trailing slashes
// …
if (content.getName().equals(path.substring(index))) {
```

though it deserves testing.